sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -88,8 +91,10 @@
   private static final String EXPORTER_THREAD_NAME =
       BatchSpanProcessor.class.getSimpleName() + "_ExporterThread";
   private final Worker worker;
-  private final Thread workerThread;
   private final boolean sampled;
+  private final ScheduledExecutorService scheduler;
+  private final ScheduledFuture<?> scheduledFutureWithFixedDelay;
+  private Future<?> imperativeSubmission;
```

Comment:
I think this reference needs to either be volatile, or implemented with an AtomicReference, since it's assignment needs to be visible across threads.

Reply:
Also, maybe renamed this to "forcedExportResult" or something along those lines? Not sure exactly what the best name is, but th
is name tripped me up for a bit.

Reply:
From what I can tell, it's pretty easy to schedule something immediately many times because of the thread safety which will be hard to solve. I think instead of a future, we can just use an `AtomicBoolean`

```java
onEnd() {
  if (forcingExport.compareAndSet(false, true)) {
    scheduler.submit(worker);
  }
}

...

Worker.run() {
...

// Last line
forcingExport.compareAndSet(true, false);
```

Reply:
That being said, this forced export logic seems suspicious. If the queue isn't keeping up, then it seems the worker will run consecutively forever ignoring `exportTimeoutMillis` completely because `worker.spansList.size() >= worker.halfMaxQueueSize` will always be true any time `forcingExport` is false in the pathological case. I think users would rather have spans dropped than this heavyweight thread running all the time?

Reply:
I think this is a good question to ask. I believe it applies to this class on the main branch as well. Maybe this should be addressed first? Can you create an issue for it?

[also, yes, just using an AtomicBoolean seems like a much simpler solution if we *do* want to force exports when the queue is full, before the interval is complete]